### ISSUE TITLE
Date of last known filing not synced properly

### ISSUE THREAD

**Issue by mlissner** [freelawproject/courtlistener#885]
We currently get the date of last known filing from the docket history report, but don't update it when we get the data from other sources. We should update the code to never allow this to be less than the latest docket entry that we have. That's obviously the date of last known filing in the system. 

Shouldn't be too hard to keep this in sync.

Also: We need to go through and update anything where this is out of sync.

**Comment by johnhawkinson** [comment]
context: RECAP
(I would still like someone to explain to me why the _DHR_ is useful.)

Where is the Date of Last Known filing displayed or used? It's not the same as the date the docket was updated that the RECAP extension adds.

>  We should update the code to never allow this to be less than the latest docket entry that we have. That's obviously the date of last known filing in the system.

Well, is it the filing date or the entry date? Did I finish dealing with the code to split them up? I can't quite remember.

What is the ramification of this proposed change?

**Comment by mlissner** [comment]
Good point about entry vs. filing date. No, we don't have both of those in the DB as of yet. We do show the date of last known filing on the docket sheet and in the API when it's available. 

> Did I finish dealing with the code to split them up?

I haven't seen any code to that end, no?

> What is the ramification of this proposed change?

I don't actually know the value of this field in the first place. Maybe the solution is to keep doing what we're doing (only update it from the DHR), and then hide it from users with an explanation in the API of how the field is only updated when we receive a DHR.

**Comment by johnhawkinson** [comment]
> > Did I finish dealing with the code to split them up?
> I haven't seen any code to that end, no?

Oops. https://github.com/johnhawkinson/juriscraper/tree/2018.06.07.date_entered
which is clearly not ready for prime time, according to my local tree, although its odd that I pushed it with a dated-branch naming convention, which I use for non-WIP work. I wonder what happened there.

> I don't actually know the value of this field in the first place. Maybe the solution is to keep doing what we're doing (only update it from the DHR), and then hide it from users with an explanation in the API of how the field is only updated when we receive a DHR.

This doesn't seem right. Is it different from what `iquery.pl` reports?

**Comment by mlissner** [comment]
I guess you're referring to this page:

![screenshot from 2018-10-24 14-00-29](https://user-images.githubusercontent.com/236970/47461278-47d46180-d795-11e8-835d-2140357d9c4f.png)

We don't crawl that or get it from extension users. The date of last filing isn't on the docket itself, so I'm pretty sure we only get it from the DHR.

**Comment by johnhawkinson** [comment]
So that's a No, it's not different. OK!

**Comment by mlissner** [comment]
Yeah, so I think I'm back to:

> I don't actually know the value of this field in the first place. Maybe the solution is to keep doing what we're doing (only update it from the DHR), and then hide it from users with an explanation in the API of how the field is only updated when we receive a DHR.

**Comment by johnhawkinson** [comment]
`Objection, Your Honor, compound!`

But don't we know we're going to have the extension parse and send the `iquery` data soon?

That's a little different from saying there is value though.

**Comment by mlissner** [comment]
I mean, maybe? But until then, the field is wrong 99% of the time and of little value...almost all of the time?

**Comment by mlissner** [comment]
I've heard from a couple people now that this field confuses them since it's not synced with the latest data in the docket entries. We need to get it syncing properly. The iquery thing is set up these days, but it doesn't run on every docket every day, so it's still often wrong.

**Comment by ikeboy** [comment]
Would it be tractable to run it on all dockets that have had an update in the last day? What about once a week for dockets active in the last week, which cuts down on active dockets by a factor of up to 7?

**Comment by mlissner** [comment]
I sort of doubt it. There are something like 150k filings/day. Maybe it's doable, but it'd be a push and I'd rather save our scraping allowance for doing mobile queries, really.

**Comment by johnhawkinson** [comment]
Honestly, we should consider the opposite: Stop updating this field for RSS, only update it when someone runs a real docket report.

That might imply splitting this into separate fields (date of last docket upload, date of last document upload, date of last RSS update).

Especially for courts with comprehensive RSS, the current system means the date becomes useless for my purpose, which is telling me how up-to-date the docket report is with respect to the Court, i.e. deciding what date range to run a docket report for. Perhaps other people use this information differently, though.

**Comment by mlissner** [comment]
What I've heard from folks is that they are confused when they see this field on dockets (or elsewhere, like favorites), because it's often just obviously wrong. 

I think we need to fix that though by showing something better in the extension. For example, what if it queried the most recent docket entry? I think we could get that quickly at scale.

**Comment by ikeboy** [comment]
Well, in favorites it should be correct after the first day.

What about kicking off a scrape immediately on docket page views that haven't been scraped in the last week, and not showing the attribute until after this scrape is complete (could load it with async call that takes some time to return?)

**Comment by johnhawkinson** [comment]
Again, what do people use this for?
I don't need CL to tell me the latest date in PACER. PACER will tell me that for free.
I want CL/RECAP to make my life more efficient.
For a case I'm willing to pay for that's usually:

. Run the docket report the first time.
. Time passes.
. Find out there are new filings (via rss, NEF, iquery, just guessing, whatever).
. Go to run the docket report again — how do I figure out what date to start, so I'm not paying for the whole thing just to pick up 2 new entries? It used to be that I could use the last update date from RECAP — although it could be unreliably if somebody else came along and, say, ran the report for 2016 yesterday instead of running the report for yesterday, for the most part it tells me what I can use as the start range for the next report.

This is now totally broken.

I'm not sure what other people are using this feature for.
If you want to tell me that I'm an outlier and nobody else does it this way, well, OK. But tell me that if that' what you think.

What is _your_ use case?

**Comment by mlissner** [comment]
Seems like what you want is the date (and possibly the entry number) of the most recent entry in CL to show up in PACER, right? 

I think other people use it in the opposite direction. They're looking at their faves or a docket and they're trying to learn if it changed since they last looked, and whether they should purchase it or whatever.

That's what I've heard, anyway.

**Comment by mlissner** [comment]
> What about kicking off a scrape immediately on docket page views that haven't been scraped in the last week, and not showing the attribute until after this scrape is complete (could load it with async call that takes some time to return?)

We could put a button next to the field to update it if people click it? Seems easier somehow and would solve some of the use case issues.

**Comment by ikeboy** [comment]
Yeah, a button clearly signals "this might be outdated"

**Comment by johnhawkinson** [comment]
> Seems like what you want is the date (and possibly the entry number) of the most recent entry in CL to show up in PACER, right?

Yes, but the entry number is a trap, so don't use it. It will cause you to miss entries that are unnumbered in the courts that use them.

> I think other people use it in the opposite direction. They're looking at their faves or a docket and they're trying to learn if it changed since they last looked, and whether they should purchase it or whatever.

Like I said, I have somewhat limited sympathy for them because they can find out this information in a different way, and the system never really supported that idea. (On the contrary, though, I can also find out my information in a different way).

> Yeah, a button clearly signals "this might be outdated"

I'm not sure what this is directed to. If me, well, sure it might be outdated, but it used to work 98% of the time, and now it's useless.

**Comment by mlissner** [comment]
OK, so sounds like we have consensus:

1. Update the last filed date field so it's never out of date with the entries (at worst), and update it via iquery via current methods (at best). This is where this bug began.

1. Show the most date of the most recent entry in CL to the extension. 

1. A button in the front end that users can use to refresh the last filed date would be good.

So...I'm going to continue with the original goal of this bug to get this field updated more consistently. The other two things can and probably should be spun into separate issues.
